core/storage/translation/gae_models.py

```diff
@@ -0,0 +1,137 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for machine translation."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+import utils
+
+(base_models,) = models.Registry.import_models(
+    [models.NAMES.base_model])
+
+datastore_services = models.Registry.import_datastore_services()
+
+
+class MachineTranslatedTextModel(base_models.BaseModel):
+    """Model for storing machine translations"""
+
+    # The untranslated text.
+    origin_text = datastore_services.StringProperty(required=True)
```

Comment:
I would expect origin_text and translated_text to either both be StringProperty or both be TextProperty. Also, the indexing requirements should be the same for both. Could you explain the discrepancy?

Note that StringProperty is for strings < 1500 chars and TextProperty is for strings greater than that. The latter cannot be indexed.

(I suspect you'll go with TextProperty. In that case, though, perhaps adding an extra StringProperty indexed field for the hashed_origin_text would be sensible.)

Reply:
This was a vestigial mistake, I changed it to TextProperty with indexed=False.
Do you think the hashed_origin_text is needed? What use cases were you thinking of?

Reply:
Yup, I think it would be useful. In the future, we might want to query for all translations for a particular string (in different languages); having such a field would let us hash that string and retrieve all translations for it.

Reply:
Makes sense! Fixed.